hypothesis-python/src/hypothesis/database.py

```diff
@@ -215,3 +228,82 @@ def delete(self, key: bytes, value: bytes) -> None:
             os.unlink(self._value_path(key, value))
         except OSError:
             pass
+
+
+class ReadOnlyDatabase(ExampleDatabase):
+    """A wrapper to make the given database read-only.
+
+    Note that this disables Hypothesis' automatic discarding of stale examples.
+    It is designed to allow local machines to access a shared database (e.g. from CI
+    servers), without propagating changes back from a local or in-development branch.
+    """
+
+    def __init__(self, db: ExampleDatabase) -> None:
+        assert isinstance(db, ExampleDatabase)
+        self._wrapped = db
+
+    def __repr__(self) -> str:
+        return f"ReadOnlyDatabase({self._wrapped!r})"
+
+    def fetch(self, key: bytes) -> Iterable[bytes]:
+        yield from self._wrapped.fetch(key)
+
+    def save(self, key: bytes, value: bytes) -> None:
+        pass
```

Comment:
For readonly database I would expect here some ReadOnly / NotImplementedError exception, rather than silent ok on write modifications. I guess, this implementation was with a reason, but may be notice that behaviour in docs? Or may be name JustForReadDatabase would be a bit clear? Read-only thing are quite nervous about write operations in common...

Reply:
Yeah, the idea is that you can use it in contexts where you want write/delete operations to be noops but not raise errors.  I'll explicitly document that though, thanks!